### PR TITLE
check for 'users:read' scope in scopes before making 'users.info' call

### LINKED ISSUE #17: if default_scopes doesn't have 'users.read', authentication callback doesn't work
Took forever to track this one down. I don't really need 'users:read' for my application, but my authentication wasn't working because the strategy tries to make the call to `users.info` even though it doesn't exist in the scope. pull request incoming

### PR DESCRIPTION
**Pull request by prashanthsadasivan**
Fixes #17

### CODE REVIEW

File: lib/ueberauth/strategy/slack.ex [hunk 1]
```diff
@@ -189,18 +189,24 @@ defmodule Ueberauth.Strategy.Slack do
...
-    case OAuth2.AccessToken.post(token, "/users.info", [token: token.access_token, user: auth["user_id"]], [{"Content-Type", "application/x-www-form-urlencoded"}]) do
-      {:ok, %OAuth2.Response{status_code: 401, body: _body}} ->
-        set_errors!(conn, [error("token", "unauthorized")])
-      {:ok, %OAuth2.Response{status_code: status_code, body: user}} when status_code in 200..399 ->
-        if user["ok"] do
-          put_private(conn, :slack_user, user["user"])
-        else
-          set_errors!(conn, [error(user["error"], user["error"])])
+    case "users:read" in scopes do
+      false -> conn
```
Comment (hassox) [inline lib/ueberauth/strategy/slack.ex:196]: Do we need to set an error here? I'm not sure how the rest of the module would react to not having a slack user on the conn when it tries to construct the auth struct.

**Comment by hassox** [comment]
I think this will break both the credentials and info calls if there is no user present.

**Comment by doomspork** [comment]
@prashanthsadasivan / @hassox any updates here?

**Comment by acconrad** [comment]
@prashanthsadasivan should probably just close this since #21 is merged?

**Pull request closed without merging**